modules/planning/tasks/deciders/path_bounds_decider.cc

```diff
@@ -16,19 +16,174 @@
 
 #include "modules/planning/tasks/deciders/path_bounds_decider.h"
 
+#include <algorithm>
+#include <limits>
+#include <string>
+#include <tuple>
+#include <vector>
+
+#include "modules/common/configs/vehicle_config_helper.h"
+
 namespace apollo {
 namespace planning {
 
+using apollo::common::ErrorCode;
 using apollo::common::Status;
+using apollo::common::VehicleConfigHelper;
+
+constexpr double kPathBoundsDeciderHorizon = 100.0;
+constexpr double kPathBoundsDeciderResolution = 0.1;
+constexpr double kDefaultLaneWidth = 5.0;
+constexpr double kDefaultRoadWidth = 20.0;
+constexpr double kRoadEdgeBuffer = 0.2;
 
 PathBoundsDecider::PathBoundsDecider(const TaskConfig &config)
     : Decider(config) {}
 
 Status PathBoundsDecider::Process(
     Frame *const frame, ReferenceLineInfo *const reference_line_info) {
-  // TODO(all) implement
+  // Sanity checks.
+  CHECK_NOTNULL(frame);
+  CHECK_NOTNULL(reference_line_info);
+
+  // The decided path bounds should be in the format of: (s, l_min, l_max).
+  std::vector<std::tuple<double, double, double>> path_boundaries;
+
```

Comment:
The s value can be easily obtained by using index * s_resolution. You can use a pair instead of a tuple.

Reply:
The reason why tuple is used is in case later, when considering side-pass, etc., the sequence may be disturbed. If we keep a record of "s", then we always have a chance to sort.